components/org.wso2.carbon.identity.conditional.auth.functions.analytics/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/analytics/CallAnalyticsFunctionImpl.java

```diff
@@ -213,4 +215,31 @@ public void cancelled() {
         });
         JsGraphBuilder.addLongWaitProcess(asyncProcess, eventHandlers);
     }
+
+    private Map<String, Object> getPayloadDataMap(Map<String, Object> payloadData) {
+        Map<String, Object> payloadDataMap = new HashMap<>();
+        for (Map.Entry<String, Object> entry : payloadData.entrySet()) {
```

Comment:
is this null safe? can `payloadData` be null?

Reply:
Better to handle NPE here. Addressed with ecb4acd9b9e89de443c225c4f58df10206068ab1